src/controllers/GameController.ts

```diff
@@ -23,9 +23,9 @@ class GameController {
 
         if (this.activeSquare) {
             const legalMoves = this.gameEngine.getLegalMoves(this.activeSquare);
-            const isLegalMove = legalMoves.some((move) => square?.row === move.row && square?.column === move.column);
+            const legalMove = legalMoves.find((move) => square?.row === move.row && square?.column === move.column);
```

Comment:
Możesz rzucić nieco światła na tą zmianę? Nie bardzo widzę dlaczego musimy na tym etapie znaleźć ten ruch

Reply:
a to leftover z poprzedniego podejścia :)"